src/pages/sign-in/index.tsx

```diff
@@ -2,27 +2,61 @@ import React, { useState } from "react"
 import Image from "next/image"
 import SignInTextField from "@/components/text-fields/sign-in-text-field"
 import { useRouter } from "next/router"
-import ErrorPopup from "@/components/popups/error-popup"
 import MainButton from "@/components/buttons/main-button"
 import { useSupabaseClient } from "@supabase/auth-helpers-react"
 import { Database } from "@/types/database"
 import { AuthResponse } from "@supabase/gotrue-js"
+import ErrorMsg from "@/components/error/error-msg"
+import { FormDataItem } from "@/pages/change-password"
+import { useForm } from "react-hook-form"
 
 const Index = () => {
-  const [email, setEmail] = useState<string>("")
-  const [password, setPassword] = useState<string>("")
-  const [showErrorPopup, setErrorPopup] = useState<boolean>()
+  const [showErrorMsg, setShowErrorMsg] = useState<boolean>()
   const [errorPopupText, setErrorPopupText] = useState<string>("")
   const supabase = useSupabaseClient<Database>()
-
   const router = useRouter()
+  const {
+    handleSubmit,
+    register,
+    watch,
```

Comment:
Could be nice to use watch to remove "invalid login credentials" when the user starts typing again. 

Like this:
```js
useEffect(() => {
    watch(() => setShowErrorMsg(false))
  }, [watch])
  ```